v2/internal/frontend/desktop/windows/frontend.go

```diff
@@ -689,6 +696,62 @@ func (f *Frontend) processMessage(message string) {
 	}()
 }
 
+func (f *Frontend) processMessageWithAdditionalObjects(message string, sender *edge.ICoreWebView2, args *edge.ICoreWebView2WebMessageReceivedEventArgs) {
+	if strings.HasPrefix(message, "file:") {
+		callbackID := message[5:]
+		objs, err := args.GetAdditionalObjects()
+		if err != nil {
+			f.logger.Error(err.Error())
+			return
+		}
+
+		if objs == nil {
+			return
+		}
+
+		_count, err := objs.GetCount()
+		if err != nil {
+			f.logger.Error(err.Error())
+			return
+		}
+
+		files := []File{}
+		count := *(*uint32)(unsafe.Pointer(&_count))
+		for i := uint32(0); i < count; i++ {
+			_file, err := objs.GetValueAtIndex(i)
+			if err != nil {
+				f.logger.Error("cannot get value at %d : %s", i, err.Error())
+				return
+			}
+
+			file := *(**edge.ICoreWebView2File)(unsafe.Pointer(&_file))
```

Comment:
I think the following should be equivalent and should make reasoning about the pointer conversion more understandable.
```suggestion
			file := (*edge.ICoreWebView2File)(unsafe.Pointer(_file))
```